pkg/manager/impl/util/shared.go

```diff
@@ -42,6 +45,81 @@ func GetTask(ctx context.Context, repo repositories.RepositoryInterface, identif
 	return &task, nil
 }
 
+func createDataReference(
+	ctx context.Context, identifier *core.Identifier, storagePrefix []string, storageClient *storage.DataStore) (
+	storage.DataReference, error) {
+	nestedSubKeys := []string{
+		identifier.Project,
+		identifier.Domain,
+		identifier.Name,
+		identifier.Version,
+	}
+	nestedKeys := append(storagePrefix, nestedSubKeys...)
+	return storageClient.ConstructReference(ctx, storageClient.GetBaseContainerFQN(ctx), nestedKeys...)
+}
+
+func WriteCompiledWorkflow(ctx context.Context, repo repositories.RepositoryInterface, storagePrefix []string,
```

Comment:
can we get some documentation on this public method

Reply:
done.